python/lsst/ap/pipe/apPipeParser.py

```diff
@@ -0,0 +1,450 @@
+#
+# This file is part of ap_pipe.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# salong with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+__all__ = ["ApPipeParser"]
+
+import argparse
+import fnmatch
+import os
+import re
+import shutil
+import sys
+
+import lsst.log as lsstLog
+import lsst.pex.config as pexConfig
+import lsst.daf.persistence as dafPersist
+import lsst.pipe.base as pipeBase
+
+DEFAULT_INPUT_NAME = "PIPE_INPUT_ROOT"
+DEFAULT_CALIB_NAME = "PIPE_CALIB_ROOT"
+DEFAULT_OUTPUT_NAME = "PIPE_OUTPUT_ROOT"
+
+
+class ApPipeParser(pipeBase.ArgumentParser):
+    """Custom argument parser to handle multiple input repos.
+    """
```

Comment:
Actually, it's exactly the same amount of work, less the overhead of an RFC and the effort of making sure the solution is generic.

I agree that putting this functionality in `pipe_base` would be a better solution in the long run. However, I'm not comfortable having this ticket, which blocks all `ap_pipe` work, be itself blocked on the discussion, implementation, and review of a change to `pipe_base`, especially if we do get significant (and, I agree, unreasonable) pushback as predicted. I can't imagine the whole process taking less than two weeks.

Given that I already have a workaround, and that it would be easy to delete it once a central solution is available (remove all but `__init__`, and express `--template` in terms of `addRepoArgument`; no interface changes needed), I'd prefer to commit the existing code _and then_ push for changes to `pipe_base`.